### PR TITLE
Implement redux state for userSession

### LINKED ISSUE #3: Move userSession to redux
We want the userSession to be reusable and modularised within the application.

Add react redux to the app and move the userSession within it.

### PR DESCRIPTION
**Pull request by amalexandrescu**
This resolves #3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,8 +21,11 @@
     "@use-it/interval": "^1.0.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-redux": "^8.0.5",
     "react-scripts": "5.0.1",
     "react-virtuoso": "^4.2.1",
+    "redux-persist": "^6.0.0",
```
Comment (Alexandrescu) [inline package.json]: Not yet needed. Please remove.

File: package.json [hunk 2]
```diff
@@ -20,8 +21,11 @@
...
+    "redux-persist-transform-encrypt": "^5.0.0",
```
Comment (Alexandrescu) [inline package.json]: So is this one.

File: src/components/Authenticate.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { useAppSelector } from '../redux/store';
+
+const Authenticate = () => {
+  const userSession = useAppSelector((state) => state.userState.userSession);
+  return (
+    <Connect
+      authOptions={{
+        appDetails: {
+          name: 'Stacks React Template',
+          // todo:
```
Comment (Alexandrescu) [inline src/components/Authenticate.tsx:15]: @BowTiedDeployer is this a todo? Then we can create a ticket.

File: src/components/ConnectWallet.tsx [hunk 4]
```diff
@@ -2,34 +2,45 @@ import { AppConfig, showConnect, UserSession } from '@stacks/connect';
...
-      window.location.reload();
-    },
-    userSession,
-  });
-};
-
-const disconnect = () => {
-  userSession.signUserOut('/');
-};
+// const authenticate = () => {
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx]: Cleanup

File: src/components/ConnectWallet.tsx [hunk 5]
```diff
@@ -2,34 +2,45 @@ import { AppConfig, showConnect, UserSession } from '@stacks/connect';
...
   currentTheme: string;
 }
 
 const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
+  const userSession = useAppSelector((state) => state.userState.userSession);
+  const dispatch = useAppDispatch();
+
+  const disconnect = () => {
+    dispatch(disconnectAction());
+    // userSession.signUserOut('/');
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx]: Cleanup

File: src/components/ConnectWallet.tsx [hunk 6]
```diff
@@ -2,34 +2,45 @@ import { AppConfig, showConnect, UserSession } from '@stacks/connect';
...
+//     userSession,
+//   });
+// };
 
 interface ConnectWalletProps {
   currentTheme: string;
 }
 
 const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
+  const userSession = useAppSelector((state) => state.userState.userSession);
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx]: Let's create a selector and use it everywhere.

File: src/index.tsx [hunk 7]
```diff
@@ -7,27 +7,34 @@ import { Connect } from '@stacks/connect-react';
...
-      >
-        <MainPage />
-      </Connect>
-    </React.StrictMode>
-  </BrowserRouter>
+  <Provider store={store}>
+    <BrowserRouter>
+      <React.StrictMode>
+        <Authenticate />
+        {/* <Connect
```
Comment (Alexandrescu) [inline src/index.tsx]: Clenaup

File: src/pages/Home.tsx [hunk 8]
```diff
@@ -52,9 +52,9 @@ const ContractCallGm = () => {
         senderAddress: userAddress,
       };
 
-      const result = await callReadOnlyFunction(options);
+      // const result = await callReadOnlyFunction(options);
```
Comment (Alexandrescu) [inline src/pages/Home.tsx]: Cleanup

File: src/redux/actions/index.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+export const CONNECT_USER_SESSION = 'CONNECT_USER_SESSION';
+
+export const disconnectAction = () => {
+  return { type: DISCONNECT_USER_SESSION };
+};
+
+export const connectAction = () => {
+  return { type: CONNECT_USER_SESSION };
+
+  // return async (dispatch: AppDispatch, getState: any) => {
```
Comment (Alexandrescu) [inline src/redux/actions/index.ts]: Cleanup

**Review by Alexandrescu — CHANGES_REQUESTED** [review]

File: src/components/Authenticate.tsx (hunk 3 above)
Comment (BowTiedDeployer) [reply src/components/Authenticate.tsx:15]: is was from the template, it displays the name of the app and its icon when authenticating.
we'll need an icon to be displayed and the name can be `decentralized mining pool`.

**Review by Alexandrescu — APPROVED** [review]

**Review by Alexandrescu — APPROVED** [review]

**Pull request merged**